node/pod.go

```diff
@@ -226,78 +226,104 @@ func (pc *PodController) updatePodStatuses(ctx context.Context, q workqueue.Rate
 
 	for _, pod := range pods {
 		if !shouldSkipPodStatusUpdate(pod) {
-			enqueuePodStatusUpdate(ctx, q, pod)
+			pc.fetchPodStatusFromProvider(ctx, q, pod)
```

Comment:
I prefer keeping "enqueue" call here and have pod status from provider fetching happening in a new line just before calling enqueue, effectively meaning two calls in this block.
The reason is that by reading this code, I now have to forcibly read `fetchPodStatusFromProvider` to understand that we may be interacting with the queue.

Reply:
Split.